XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/Hands/HandRay.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using XRTK.Definitions.Utilities;
+using XRTK.Utilities.Physics;
+
+namespace XRTK.Providers.Controllers.Hands
+{
+    public class HandRay
+    {
+        // Constants from Shell Implementation of hand ray
+        private const float DynamicPivotBaseY = -0.1f, DynamicPivotMultiplierY = 0.65f, DynamicPivotMinY = -0.6f, DynamicPivotMaxY = -0.2f;
+        private const float DynamicPivotBaseX = 0.03f, DynamicPivotMultiplierX = 0.65f, DynamicPivotMinX = 0.08f, DynamicPivotMaxX = 0.15f;
+        private const float HeadToPivotOffsetZ = 0.08f;
+        private readonly float CursorBeamBackwardTolerance = 0.5f;
+        private readonly float CursorBeamUpTolerance = 0.8f;
+
+        private GenericStabilizer stabilizer = new GenericStabilizer();
+        private Vector3 palmNormal;
+        private Vector3 headForward;
+
+        private Ray ray = new Ray();
+        public Ray Ray
+        {
+            get
+            {
+                ray.origin = stabilizer.StablePosition;
+                ray.direction = stabilizer.StableRay.direction;
+                return ray;
+            }
+        }
+
+        public bool ShouldShowRay
+        {
+            get
+            {
+                if (headForward.magnitude < Mathf.Epsilon)
+                {
+                    return false;
+                }
+
+                bool valid = true;
+                if (CursorBeamBackwardTolerance >= 0)
```

Comment:
This whole math check is convoluted.

In my experience, the one I do in AltHandRay for MRTK is sufficient, and easier to read.
This is a very subjective change though.
https://github.com/microsoft/MixedRealityToolkit-Unity/pull/6973/files#diff-4e35f1fe0a91ac28a11deb3ae0504bba